zaas-client/src/main/java/org/zowe/apiml/zaasclient/service/internal/ZaasClientImpl.java

```diff
@@ -26,12 +27,17 @@
     private final PassTicketService passTickets;
 
     public ZaasClientImpl(ConfigProperties configProperties) throws ZaasConfigurationException {
+        if (configProperties.getKeyStorePath() == null) {
+            throw new ZaasConfigurationException(ZaasConfigurationErrorCodes.KEY_STORE_NOT_PROVIDED);
+        }
+
         CloseableClientProvider httpClientProvider = getTokenProvider(configProperties);
+        CloseableClientProvider httpClientProviderWithoutCert = getTokenProviderWithoutCert(configProperties, httpClientProvider);
```

Comment:
In case of HTTP , the same provider object is used, in case of HTTPS, there are 2 objects being created, is this intentional?

Reply:
Yes, it is, in the case of HTTP there is no needed to make two instances. But the purpose is the question because HTTP is not possible to use (except the theoretical configuration). It should be just for ATTLS, but then it is too difficult (maybe impossible) to establish to respect URL and using a client certificate.

Reply:
We should focus on the possibility, how to use ZAAS client with AT-TLS in the next issue. Thank you for finding that.